public/app/legacy/models/units.ts

```diff
@@ -9,16 +9,13 @@ export const units = [
   'lock_nanoseconds',
   'trace_samples',
   'exceptions',
-];
+  'nanoseconds',
+] as const;
 
-export const UnitsSchema = z.preprocess((u) => {
-  if (typeof u === 'string') {
-    if (units.includes(u)) {
-      return u;
-    }
-  }
-  return 'unknown';
-}, z.enum(['samples', 'objects', 'goroutines', 'bytes', 'lock_samples', 'lock_nanoseconds', 'trace_samples', 'exceptions', 'unknown']));
+export const UnitsSchema = z.preprocess(
```

Comment:
In the future, we should also refactor this to use the JSON data instead.

Reply:
can you either leave a comment in the code or create an issue (or both) so we can come back to this at some point?

Reply:
Sure, here it is: https://github.com/grafana/pyroscope/issues/2740